src/ReinfyTeam/ProfanityFilter/Loader.php

```diff
@@ -107,7 +107,12 @@ private function registerCommands() : void {
 	}
 
 	private function checkUpdate() : void {
-		$this->getServer()->getAsyncPool()->submitTask(new PoggitUpdateTask($this->getDescription()->getName(), $this->getDescription()->getVersion()));
+		$lang = new Language();
+		if ((bool) ($this->getConfig()->get("check-updates") ?? true)) {
```

Comment:
Why you are checking of a non-nullable function? The function will return false when it is null.